server/types/oid/iterate.go

```diff
@@ -159,6 +160,35 @@ func IterateCurrentDatabase(ctx *sql.Context, callbacks Callbacks) error {
 	return nil
 }
 
+// IterateTypes iterates over all types that are present in the pg_type table.
+func IterateTypes(ctx *sql.Context, cb func(displayType pgtypes.DoltgresType) (cont bool, err error)) error {
+	allTypes := pgtypes.GetAllTypes()
+	// Exclude all types that are not present in the pg_type table
+	for _, t := range allTypes {
+		switch t.BaseID() {
+		case pgtypes.DoltgresTypeBaseID_Null,
+			pgtypes.DoltgresTypeBaseID_Int16Serial,
+			pgtypes.DoltgresTypeBaseID_Int32Serial,
+			pgtypes.DoltgresTypeBaseID_Int64Serial:
+			continue
+		default:
+			if _, ok := t.(pgtypes.DoltgresArrayType); !ok {
+				if _, ok = t.(pgtypes.DoltgresPolymorphicType); !ok {
+					cont, err := cb(t)
+					if err != nil {
+						return err
+					}
+					if !cont {
+						return nil
+					}
+				}
+			}
+		}
+	}
+	_, err := cb(pgtypes.InternalChar)
+	return err
+}
```

Comment:
This should be a callback on a schema element, and placed within `IterateCurrentDatabase`. We should also have a `runTypes` variant that's within `RunCallback`, and a check within `runCallbackValidation` as well.